nifi-extension-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/CaptureBoxEvents.java

```diff
@@ -0,0 +1,227 @@
+@InputRequirement(Requirement.INPUT_FORBIDDEN)
+public class CaptureBoxEvents extends AbstractProcessor implements VerifiableProcessor {
+
+    public static final PropertyDescriptor MAX_MESSAGE_QUEUE_SIZE = new PropertyDescriptor.Builder()
+            .name("Max Size of Message Queue")
+            .description("The maximum size of the internal queue used to buffer events being transferred from the underlying stream to the processor. " +
+                    "Setting this value higher allows more messages to be buffered in memory during surges of incoming messages, but increases the total " +
+                    "memory used by the processor during these surges.")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("10000")
+            .required(true)
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = List.of(
+            BoxClientService.BOX_CLIENT_SERVICE,
+            MAX_MESSAGE_QUEUE_SIZE
+    );
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Events received successfully will be sent out this relationship.")
+            .build();
+
+    private static final Set<Relationship> RELATIONSHIPS = Set.of(REL_SUCCESS);
+
+    private volatile BoxAPIConnection boxAPIConnection;
+    private volatile EventStream eventStream;
+    protected volatile LinkedBlockingQueue<BoxEvent> events;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    @Override
+    public final Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        final BoxClientService boxClientService = context.getProperty(BoxClientService.BOX_CLIENT_SERVICE).asControllerService(BoxClientService.class);
+        boxAPIConnection = boxClientService.getBoxApiConnection();
+        eventStream = new EventStream(boxAPIConnection);
+
+        final int eventsCapacity = context.getProperty(MAX_MESSAGE_QUEUE_SIZE).asInteger();
+        if (events == null) {
+            events = new LinkedBlockingQueue<>(eventsCapacity);
+        }
+
+        eventStream.addListener(new EventListener() {
+            @Override
+            public void onEvent(BoxEvent event) {
+                if (!events.offer(event)) {
+                    getLogger().warn("Failed to add event to queue. Queue is full.");
+                }
+            }
+
+            @Override
+            public void onNextPosition(long position) {
+                getLogger().debug("Next position: {}", position);
+            }
+
+            @Override
+            public boolean onException(Throwable e) {
+                getLogger().warn("An error has been received from the stream.", e);
```

Comment:
Recommend tracking the `position` so that it could be logged.
```suggestion
                getLogger().warn("Event stream processing failed at position {}", position, e);
```

Reply:
We do not have the position when an exception is thrown. We have it in `onNextPosition` and it's already logged there. Or am I missing something?

Reply:
Right, the change would involve tracking the last position from `onNextPosition` in an AtomicLong, so it could be pulled and referenced in this warning. That could be helpful when debugging is not enabled.

Reply:
Yep, understood. I looked a bit more into the Box code to see how this position is tracked and I see how it can be useful in certain scenarios.